examples/app/vsgcameras/vsgcameras.cpp

```diff
@@ -24,6 +24,29 @@ class CameraSelector : public vsg::Inherit<vsg::Visitor, CameraSelector>
     vsg::ref_ptr<vsg::Camera> main_camera;
     vsg::ref_ptr<vsg::LookAt> original_viewMatrix;
     Cameras cameras;
+    int currentCameraIndex = 0;
+
+    void apply(vsg::ButtonPressEvent &buttonPress) override
```

Comment:
FYI, the VSG and the rest of eco system use "TYPE& variable" convention rather than "TYPE &variable" convention.  To me the type is TYPE& and &variable is the address.